File: day42/homework/hw.py
# 1) შექმენი სია ხილებზე და დაამატე მასში კიდევ 2 ხილი extend() ფუნქციით.

fruits = ["apple", "banana", "pumpkin"]

fruits2 = ["avokado", "orange", "coconut"]

fruits.extend(fruits2)

print(fruits)


# 2) შექმენი სია numbers და დაამატე მასში [40, 50] extend()-ით.

numbers = [1,2,3,4,5,6,7,8,9,10]

numbers2 = [40,50]

numbers.extend(numbers2)

print(numbers)

# 3) შექმენი სია names და შეაბრუნე reverse()-ით.

names = ["rezi", "goga", "gega", "dachi", "nika"]

names.reverse()

print(names)

# 4) შექმენი სია სახელად nums და დათვალე რამდენი ცალი 5 არის მასში count()-ით.

nums = [1,2,3,4,5,6,5,4,3,2,1]

print(nums.count(5))

# 5) შექმენი letters = ["a","b","a","c"] და დაბეჭდე რამდენი ცალი "a" არის ჩვენს სიაში.

letters = ["a","b","a","c"]

print(letters.count("a"))

# 6) შექმენი სია სახელად names და იპოვე "saba"-ს ინდექსი index()-ით.

names = ["rezi", "goga", "saba", "gega", "dachi", "nika"]

print(names.index("saba"))

# 7) შექმენი list = ["red","green","blue"] და იპოვე რომელ ინდექსზე დგას "blue". გამოიყენე შესაბამისი ფუნქცია.

list = ["red","green","blue"]

print(list.index("blue"))

# 8) შექმენი სია სახელად nums და დამატე მასში extend()-ით [7, 8, 9].

nums = [1,2,3,4,5,6]

nums2 = [7,8,9]

nums.extend(nums2)

print(nums)

# 9) შექმენი სია სახელად foods და დააბრუნე შებრუნებული სია.

foods = ["xinkali", "xawapuri", "kubdari", "mwvadi"]

foods.reverse()

print(foods)

# 10) შექმენი სია cities და იპოვე რომელ ინდექსზე დგას "tbilisi".

cities = ["qutaisi", "tbilisi", "rustavi", "batumi"]

print(cities)
# 11) შექმენი animals = ["cat","dog","cat","cow"] და დაითვალე ამ სიაში რამდენი "cat" არის.

animals = ["cat","dog","cat","cow"]

print(animals.count("cat"))

# 12)შექმენი სია fruits = ["apple", "banana"] და append ფუნქციით დაამატე "grape". დაბეჭდე სია.

fruits = ["apple", "banana"]

fruits.append("grape")

print(fruits)

# 13)შექმენი სია numbers = [1, 2, 3] და extend()-ით დაუმატე [4, 5]. დაბეჭდე სია.

numbers = [1, 2, 3]

num = [4, 5]

numbers.extend(num)

print(numbers)

# 14)შექმენი სია names = ["goga", "saba"] და insert()-ით ჩასვი "luka" პირველ ინდექსზე. დაბეჭდე სია.

names = ["goga", "saba"]

names.insert(1, "luka")

print(names)

# 15)შექმენი სია items = ["pen", "pencil", "eraser"] და pop()-ით წაშალე ბოლო ელემენტი; დაბეჭდე განახლებული სია.

items = ["pen", "pencil", "eraser"]

items.pop(2)

print(items)

# 16)შექმენი სია colors = ["red", "green", "blue"] და remove()-ით წაშალე "green". დაბეჭდე შედეგი.

colors = ["red", "green", "blue"]

colors.remove("green")

print(colors)

# 17)შექმენი სია foods = ["bread", "milk"]. შეამოწმე სიაში 2 ელემენტია თუ მეტი — თუ ორია, append()-ით დაამატე "cheese", შემდეგ დაბეჭდე სია, სხვა შემთხვევაში append()-ით დაამატე "meat" და დაბეჭდე სია.

foods = ["bread", "milk"]

if len(foods) >= 2:
    foods.append("cheese")
    print(foods)
else:
    foods.append("meat")
    print(foods)

# 18)შექმენი სია nums = [10, 20, 30]. მომხმარებელს შემოატანინე მთელი რიცხვი. თუ რიცხვი nums სიაშია, დაბეჭდე "Already in list", თუ არა — append()-ით დაამატე 40 და დაბეჭდე სია.

nums = [10, 20, 30]

integer = int(input("შემოიტანე მთელი რიცხვი: "))

if integer in nums:
    print("Already in list")
else:
    nums.append(40)
    print(nums)

# 19)შექმენი სია letters = ["a", "b", "c"]. მომხმარებელს შემოატანინე ასო, შემდეგ insert()-ით ჩასვი ის სიის შუაში (ცენტრალურ ინდექსზე). დაბეჭდე სია.

letters = ["a", "b", "c"]

letter = input("type any letter: ")

letters.insert(2, letter)

print(letters)

# 20)შექმენი სია values = [1, 2, 3, 4]. მომხმარებელს შემოატანინე ინდექსი. თუ ინდექსი სიის ფარგლებშია, pop()-ით ამოშალე შესაბამისი ელემენტი; თუ არა, დაბეჭდე "Index out of range". 
# ბოლოს დაბეჭდე სია.

values = [1, 2, 3, 4]

index = int(input("type index: "))

if index <= len(values):
    values.pop(index)
else:
    print("Index out of range")

print(values)

# 21)შექმენი სია pets = ["cat", "dog", "hamster"].  მომხმარებელს შემოატანინე შინაური ცხოველის სახელი. თუ იგი არის სიის შიგნით, remove()-ით ამოშალე და დაბეჭდე "Removed", 
# თუ არა — დაბეჭდე "Not found" და სია უცვლელი დატოვე; საბოლოოდ დაბეჭდე სია.

pets = ["cat", "dog", "hamster"]

user_pet = input("Enter pet name: ")

if user_pet in pets:
    pets.remove(user_pet)
    print("Removed")
else:
    print("Not found")

print(pets)

# 22)შექმენი სია a = [5, 5, 7]. მომხმარებელს შემოატანინე რიცხვი. თუ რიცხვი არის სიის ელემენტი, დაბეჭდე რამდენჯერ არის სიაში - count() ფუნქციის გამოყენებით. 
# სხვა შემთხვევაში append()-ით ჩასვი ის სიაში და დაბეჭდე სია.

a = [5, 5, 7]

user_num = int(input("შეიტანეთ რიცხვი: "))

if user_num in a:
    print(a.count(user_num))
else:
    a.append(user_num)
    print(a)

# 23)შექმენი სია queue = ["first", "second"].  მომხმარებელს შემოატანინე ახალი ელემენტი და insert()-ით ჩასვი სიის დასაწყისში. შემდეგ if-ით შეამოწმე სიის სიგრძე — თუ უფრო დიდია 5-ზე, pop()-ით ამოშალე ბოლო ელემენტი; ბოლოს დაბეჭდე სია, თუ არ არის 5-ზე მეტი დაბეჭდე შებრუნებული სია.

queue = ["first", "second"]

user = input("შეიყვანე ახალი ელემენტი: ")

queue.insert(0, user)

if len(queue) > 5:
    queue.pop()
    print(queue)
else:
    print(list(reversed(queue)))

# 24)შექმენი სია nums = [2, 4, 6].  მომხმარებელს შემოატანინე რიცხვი. თუ რიცხვი დადებითია, append()-ით დაამატე; თუ 0-ია ან ნაკლებია ნულზე, დაბეჭდე "Only positive allowed". 
# ბოლოს დაბეჭდე სია.

nums = [2, 4, 6]

user_number = int(input("Enter a number: "))

if user_number > 0:
    nums.append(user_number)
else:
    print("Only positive allowed")

print(nums)

# 25) შექმენი სია mix = ["x", "y", "z"]. extend()-ით დაუმატე [1, 2, 3]. შემდეგ მომხმარებელს შემოატანინე ასო; თუ ეს ასო არის სიაში, remove()-ით წაშალე პირველად როცა შეგხვდება და დაბეჭდე "Removed", თუ არა — დაბეჭდე "No such element". ბოლოს დაბეჭდე სია.

mix = ["x", "y", "z"]

num = [1, 2, 3]

mix.extend(num)

letter = input("Enter a letter: ")

if letter in mix:
    mix.remove(letter)
    print("Removed")
else:
    print("No such element")

print(mix)

File: level 96/homework/hw.js
/*1)let numbers = [12, 45, 7, 23, 89, 34, 16, 50];
პირველი ელემენტი შეცვალე 100-ით
ბოლო ელემენტი შეცვალე 200-ით
მესამე ელემენტი გაზარდე 10-ით
მეხუთე ელემენტი გაანახევრე
მეოთხე და მეშვიდე ელემენტები ადგილებზე გაცვალე

საბოლოოდ გამოიტანე განახლებული მასივი.*/ 

let numbers = [12, 45, 7, 23, 89, 34, 16, 50]

numbers[0] = 100
numbers[7] = 200
numbers[2] = numbers[2] + 10
numbers[4] = numbers[4] / 2
numbers[3] = 16
numbers[6] = 23

console.log(numbers)

/*2)კონკრეტული ელემენტების პოვნა და შეცვლა
let numbers = [15, 8, 42, 8, 31, 42, 19, 8];

მოთხოვნები:

პირველი 8 შეცვალე 80-ით
პირველი 42 შეცვალე 420-ით
ბოლო 8 შეცვალე 800-ით
მეორე ელემენტი გაზარდე 5-ით
ბოლოს დაბეჭდე მასივი*/ 

let numbers1 = [15, 8, 42, 8, 31, 42, 19, 8]

numbers1[1] = 80
numbers1[2] = 420
numbers1[7] = 800
numbers1[1] = 80 + 5

console.log(numbers1)

/*3)let fruits = ["apple", "banana", "orange", "kiwi", "mango", "peach"];

შექმენი ცვლადი:

let index = 3;

პროგრამამ უნდა:

გამოიტანოს არჩეულ ინდექსზე არსებული ხილი;
შეცვალოს ის "watermelon"-ით;
ისევ გამოიტანოს ეს ელემენტი;
გამოიტანოს საბოლოო მასივი.

შემდეგ შეცვალე index სხვადასხვა მნიშვნელობით და შეამოწმე შედეგი.*/ 

let fruits = ["apple", "banana", "orange", "kiwi", "mango", "peach"]
let index = 3

fruits[index] = "watermelon"
console.log(fruits)

/*4)შექმენი სახელების სია ,ასევე შექმენი ცვლადი სადაც მომხმარებელს შემოატანინებ რაიმე რიცხვს
(დაგჭირდებათ შემოტანილი მნშვენლობის რიცხვად გარდაქმნა) და გამოიტანეთ ამ რიცხვზე მდგომი ელემენტი კონსოლში*/ 

let names = ["rezi", "gega", "dachi"]

let num = "1"

num = Number(num)
console.log(names[num])

/*5)ელემენტის ჩანაცვლება მისი პოზიციის მიხედვით
let students = [
    "Giorgi",
    "Nika",
    "Saba",
    "Luka",
    "Dato",
    "Ana"
];

მოცემულია:

let position = 4;
let newName = "Goga";

position აღნიშნავს ადამიანურ პოზიციას — ანუ 1 ნიშნავს პირველ ელემენტს.

დავალება:

position-ის მიხედვით იპოვე შესაბამისი ელემენტი;
ჩაანაცვლე newName-ით;
თუ ასეთი პოზიცია არ არსებობს, გამოიტანე "Invalid position".*/ 

let students = ["Giorgi","Nika","Saba","Luka","Dato","Ana"]
let position = 4
let newName = "Goga"

if(position <= students.length){
    students[position - 1] = newName
}else{
    console.log("Invalid position")
}

console.log(students)

/*6). ელემენტის წაშლა და მის ადგილას ახლის ჩასმა
let colors = ["red", "blue", "green", "yellow", "black", "white"];

დავალება:

წაშალე "green";
მის ადგილას ჩასვი "purple";
"black" შეცვალე "pink"-ით;
პირველი ელემენტი გადაიტანე ბოლოში.*/ 

let colors = ["red", "blue", "green", "yellow", "black", "white"]

colors[2] = "purple"
colors[4] = "pink"
colors[6] = "red"
colors = colors.slice(1)
console.log(colors)

/*7)let numbers = [5, 10, 15, 20, 25, 30, 35, 40];

უნდა მიიღო:

[5, 100, 150, 200, 250, 300, 35, 40]

ანუ მეორე ელემენტიდან მეექვსე ელემენტამდე თითოეული ელემენტი გაამრავლე 10-ზე.*/ 

let numbers2 = [5, 10, 15, 20, 25, 30, 35, 40]

numbers2[1] = numbers2[1] * 10
numbers2[2] = numbers2[2] * 10
numbers2[3] = numbers2[3] * 10
numbers2[4] = numbers2[4] * 10
numbers2[5] = numbers2[5] * 10

console.log(numbers2)

/*8)let numbers = [10, 20, 30, 40, 50, 60, 70, 80];

პროგრამამ უნდა შეასრულოს:

კენტ ინდექსებზე მყოფი ელემენტები გაზარდოს 5-ით;
ლუწ ინდექსებზე მყოფი ელემენტები გაამრავლოს 2-ზე.*/ 



